### PR TITLE
notify-api-91 sanitize SQL inputs against injection

### PR DESCRIPTION
**Pull request by terrazoon**

### CODE REVIEW

**Review by ccostino — COMMENTED** [review]
Thanks, @terrazoon!  This is definitely good to tackle!

However, I'm concerned that this won't fully solve the issue because while it's simulating a parameterized query, it isn't an actual parameterized query; it's just still a formatted string, only using `format` instead of the `f''` (f-string) syntax.

If we truly want to avoid SQL injection, I think we'll have to use a combination of SQLAlchemy itself along with Alembic.  Alembic's documentation on the [`execute`](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.execute) method states that a query is converted to a [SQLAlchemy `TextClause` construct](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.text), which serves as the basis for constructing a parameterized query with bound parameters.

The documentation goes further with a guide on [how to use Textual SQL](https://docs.sqlalchemy.org/en/14/core/tutorial.html#using-textual-sql) to achieve working with bound parameters.

In the case of migrations, the actual query execution is done within Alembic's framework and not SQLAlchemy directly, but Alembic's execute can either take a plain string or "...it can be any kind of Core SQL Expression construct" - meaning we ought to be able to construct a SQLAlchemy-based query that properly accounts for bound parameters and pass that in for execution instead of a plain string.

There is [a note in the parameters list for Alembic's `execute` method](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.execute.params.sqltext) about the use of bound parameters:

>**Note** when passing a plain string, the statement is coerced into a [sqlalchemy.sql.expression.text()](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.text) construct. This construct considers symbols with colons, e.g. :foo to be bound parameters. To avoid this, ensure that colon symbols are escaped, e.g. \:foo.

So there may be some trial and error in figuring out the proper formatting and escaping.

However, were these migrations auto-generated by Alembic, or where they created manually?  If these were auto-generated to begin with, this may all be overkill to try and work in.

**Comment by terrazoon** [comment]
I've converted this initial set of files from using f-strings to using text().  However, there will be another large set of files that are currently using format() that will need changing, so this is still in progress.  For some reason the "convert to draft" button is not working for me.

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon! 🎉 

I went through all of these and all looks fine to me, I noted just one small thing in migration `0021` but not a showstopper, so approving.

File: migrations/versions/0021_add_delivered_failed_counts.py [hunk 1]
```diff
@@ -21,19 +23,22 @@ def upgrade():
     conn = op.get_bind()
     results = conn.execute("select distinct job_id from notifications")
     res = results.fetchall()
-
+    conn = op.get_bind()
```
[the remark this answers is not in the document]
Comment (terrazoon) [reply migrations/versions/0021_add_delivered_failed_counts.py]: Fixed!

**Pull request merged**